jdaviz/configs/imviz/plugins/links_control/links_control.py

```diff
@@ -19,13 +20,39 @@ class LinksControl(TemplateMixin):
     wcs_use_fallback = Bool(True).tag(sync=True)
     wcs_use_affine = Bool(True).tag(sync=True)
 
-    def vue_do_link(self, *args, **kwargs):
+    linking_in_progress = Bool(False).tag(sync=True)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.hub.subscribe(self, LinkUpdatedMessage,
+                           handler=self._on_link_updated)
+
+    def _on_link_updated(self, msg):
+        self.link_type = {'pixels': 'Pixels', 'wcs': 'WCS'}[msg.link_type]
+        self.wcs_use_fallback = msg.wcs_use_fallback
+        self.wcs_use_affine = msg.wcs_use_affine
+
+    @observe('link_type', 'wcs_use_fallback', 'wcs_use_affine')

```

Comment:
This still blows my mind. On message, it updates traitlet, and this other thing observes the traitlet and call something that emits the message that would have updated it... but if you say it works, I'll have to believe you.

Maybe you want to demo this during review and see how the POs like it.

Reply:
I think the observes only trigger on a value _change_, so setting to an existing value is ignored automatically, and means we don't need to worry about infinite loops!

Reply:
ah, but there is one case where the linking could still get called twice, I'm fixing that now (even though it seemed to not cause any significant performance hit - glue must be doing something clever under-the-hood in `set_links`).